rfcs/rfcOVID-0019.md

```diff
@@ -0,0 +1,329 @@
+
+There are many exporter modules available. The core `Exporter` is the best
+known, and has the curious feature of allowing procedural modules to use
+inheritance to implement exporting via package variables. It has a clumsy
+interface and many alternatives arose:
+
+* [Sub::Exporter](https://metacpan.org/pod/Sub::Exporter)
+* [Exporter::Lite](https://metacpan.org/pod/Exporter::Lite)
+* [Exporter::Easy](https://metacpan.org/pod/Exporter::Easy)
+* [Exporter::Tiny](https://metacpan.org/pod/Exporter::Tiny)
+
+And more, including many modules which manually implement exporting directly
+in the `import` method.
+
+There are also many attribute-based exporters:
+
+* [Exporter::Simple](https://metacpan.org/pod/Exporter::Simple)
+* [Perl6::Export](https://metacpan.org/pod/Perl6::Export) (not really attributes)
+* [Perl6::Export::Attrs](https://metacpan.org/pod/Perl6::Export::Attrs)
+* [Exporter::Attributes](https://metacpan.org/pod/Exporter::Attributes)
+
+And so on.
+
+## Specification
+
+### Usage
+
+Because lower-case attribute names are reserved for Perl's future expansion,
+we do not need a feature guard. Instead, we can simply `use v5.XX` to pull in
+the new feature. However, even that might not be needed if they're using a
+future Perl since it doesn't conflict with anything default.
+
+```perl
+package My::Module {
+    use v5.38;
+
+    sub bar  :export           {...} # @EXPORT_OK
+    sub baz  :export(strings)  {...} # %EXPORT_TAGS
+    sub bay  :export(strings)  {...} # %EXPORT_TAGS
+    sub quux :export(DEFAULT)  {...} # @EXPORT
+    sub whee                   {...} # not exported
+}
+```
+
+The above behaves similarly to:
+
+```perl
+package My::Module {
+    use base 'Exporter';
+
+    our @EXPORT    = qw(quux);
+    our @EXPORT_OK = qw(
+        bar
+        baz
+        bay
+    )
+    our %EXPORT_TAGS = qw(
+        'all'     => \@EXPORT_OK,
+        'strings' => qw(bay baz),
+    );
+
+    sub bar  {...}
+    sub baz  {...}
+    sub bay  {...}
+    sub quux {...}
+    sub whee {...}
+}
+```
+
+The requested functions are imported lexically via the
+`builtin::export_lexically` function, rather than appearing in the package
+symbol table. This means that code within that lexical scope can see and call
+them as normal, but they are not visible from outside the scope to which they
+are exported, such as from other callers or as object or class methods. This
+removes the need for `namespace::clean` and related modules.
+
+Upper-case `:export` attribute arguments are reserved for Perl to avoid
+clashing with user-defined tags. This is because in this author's experience,
+export tags are _usually_ lower-case.
+
+### Interface
+
+The various incantations of `:export` should populate the respective
+`@EXPORT`, `@EXPORT_OK`, and `%EXPORT_TAGS` package variables to maintain a
+consistent API with the `Exporter` module.
+
+The grammar for the tag is:
+
+    <export>  ::= ':export' [ <options> ]
+    <options> ::=  '(' ( 'DEFAULT' | <tags> ) ')'
```

Comment:
This grammar doesn't allow listing an export in a tag and in DEFAULT. That should be allowed.

Reply:
Fixed.